Cannot read property 'QueryBuilder' of undefined

Compiling and running a simple typescript example results in `TypeError: Cannot read property 'QueryBuilder' of undefined`.

# Environment

Knex version: 0.95.0
Node version: 14.16.0
Typescript version: 4.2.3

# Steps to reproduce:

```
$ yarn init -y
$ yarn add typescript knex @types/node
$ tsc --init
```

Create a test.ts file with:
```ts
import {Knex} from 'knex'
console.log(Knex.QueryBuilder)
```

Then
```
$ tsc test.ts
$ node test.js
```

This yields the following error:
```
console.log(knex_1.Knex.QueryBuilder);
                        ^

TypeError: Cannot read property 'QueryBuilder' of undefined
    at Object.<anonymous> (/xxx/test.js:4:25)
    at Module._compile (internal/modules/cjs/loader.js:1063:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1092:10)
    at Module.load (internal/modules/cjs/loader.js:928:32)
    at Function.Module._load (internal/modules/cjs/loader.js:769:14)
    at Function.executeUserEntryPoint [as runMain] (internal/modules/run_main.js:72:12)
    at internal/main/run_main_module.js:17:47
```

The `Knex` variable is not exported from the package, only `knex`.
A simple fix could be to just add `knex.Knex = knex` to knex.js ?

But that won't even compile

```
error TS2339: Property 'QueryBuilder' does not exist on type '<TRecord extends {} = any, TResult = unknown[]>(config: string | Config<any>) => Knex<TRecord, TResult>'
```

--- a/lib/knex-builder/Knex.js
+++ b/lib/knex-builder/Knex.js
@@ -6,7 +6,7 @@ const makeKnex = require('./make-knex');
 const { KnexTimeoutError } = require('../util/timeout');
 const { resolveConfig } = require('./internal/config-resolver');
 
-function Knex(config) {
+function knex(config) {
   const { resolvedConfig, Dialect } = resolveConfig(...arguments);
 
   const newKnex = makeKnex(new Dialect(resolvedConfig));
@@ -17,15 +17,15 @@ function Knex(config) {
 }
 
 // Expose Client on the main Knex namespace.
-Knex.Client = Client;
+knex.Client = Client;
 
-Knex.KnexTimeoutError = KnexTimeoutError;
+knex.KnexTimeoutError = KnexTimeoutError;
 
-Knex.QueryBuilder = {
+knex.QueryBuilder = {
   extend: function (methodName, fn) {
     QueryBuilder.extend(methodName, fn);
     QueryInterface.push(methodName);
   },
 };
 
-module.exports = Knex;
+module.exports = knex;

--- a/test-tsd/querybuilder.test-d.ts
+++ b/test-tsd/querybuilder.test-d.ts
@@ -1,4 +1,4 @@
-import { Knex, knex } from '../types';
+import { knex } from '../types';
 import { expectType } from 'tsd';
 import { clientConfig } from './common';
 
@@ -18,7 +18,7 @@ declare module '../types' {
   }
 }
 
-Knex.QueryBuilder.extend('customSelect', function (value: number) {
+knex.QueryBuilder.extend('customSelect', function (value: number) {
   return this.select(this.client.raw(`${value} as value`));
 });
 

--- a/test-tsd/types.test-d.ts
+++ b/test-tsd/types.test-d.ts
@@ -5,7 +5,7 @@ import knexDefault, { Knex, knex } from '../types';
 import * as knexStar from '../types';
 import knexCjsImport = require('../');
 import QueryBuilder = Knex.QueryBuilder;
-import KnexTimeoutError = Knex.KnexTimeoutError;
+import KnexTimeoutError = knex.KnexTimeoutError;
 
 const knexCjs = require('../knex');
 const { knex: knexCjsNamed } = require('../knex');
@@ -16,8 +16,8 @@ expectType<Knex<any, unknown[]>>(knexStar.default({}));
 expectType<Knex<any, unknown[]>>(knexStar.knex({}));
 expectType<Knex<any, unknown[]>>(knexCjsImport.default({}));
 expectType<Knex<any, unknown[]>>(knexCjsImport.knex({}));
-expectType<KnexTimeoutError>(new KnexTimeoutError());
-expectType<KnexTimeoutError>(new KnexTimeoutError());
+expectType<KnexTimeoutError>(new knex.KnexTimeoutError());
+expectType<KnexTimeoutError>(new knex.KnexTimeoutError());
 
 // eslint-disable-next-line
 expectType<any>(knexCjs({}));

--- a/types/index.d.ts
+++ b/types/index.d.ts
@@ -397,6 +397,20 @@ export declare function knex<TRecord extends {} = any, TResult = unknown[]>(
   config: Knex.Config | string
 ): Knex<TRecord, TResult>;
 
+export declare namespace knex {
+  class QueryBuilder {
+    static extend(
+      methodName: string,
+      fn: <TRecord extends {} = any, TResult = unknown[]>(
+        this: Knex.QueryBuilder<TRecord, TResult>,
+        ...args: any[]
+      ) => Knex.QueryBuilder<TRecord, TResult>
+    ): void;
+  }
+
+  export class KnexTimeoutError extends Error {}
+}
+
 export declare namespace Knex {
   //
   // Utility Types
@@ -2350,10 +2364,6 @@ interface MsSqlConnectionConfigBase {
     constraintName?: string;
   }
 
-  //
-  // Clients
-  //
-
   class Client extends events.EventEmitter {
     constructor(config: Config);
     config: Config;
@@ -2408,18 +2418,6 @@ interface MsSqlConnectionConfigBase {
     assertCanCancelQuery(): void;
     cancelQuery(): void;
   }
-
-  class QueryBuilder {
-    static extend(
-      methodName: string,
-      fn: <TRecord extends {} = any, TResult = unknown[]>(
-        this: QueryBuilder<TRecord, TResult>,
-        ...args: any[]
-      ) => QueryBuilder<TRecord, TResult>
-    ): void;
-  }
-
-  export class KnexTimeoutError extends Error {}
 }
 
 export default knex;

